controllers/rclone_mover_test.go

```diff
@@ -0,0 +1,671 @@
+		Expect(k8sClient.Delete(ctx, namespace)).To(Succeed())
+	})
+	JustBeforeEach(func() {
+		// at least the srcPVC & rcloneSecret should be expected to start at this point
+		Expect(k8sClient.Create(ctx, srcPVC)).To(Succeed())
+		Expect(k8sClient.Create(ctx, rcloneSecret)).To(Succeed())
+	})
+	When("Components are expected to start", func() {
+		JustBeforeEach(func() {
+			// source pvc comes up
+			Expect(k8sClient.Create(ctx, rs)).To(Succeed())
+			// wait for the ReplicationSource to actually come up
+			Eventually(func() error {
+				inst := &scribev1alpha1.ReplicationSource{}
+				return k8sClient.Get(ctx, utils.NameFor(rs), inst)
+			}, maxWait, interval).Should(Succeed())
+		})
+
+		When("ReplicationSource is provided with an Rclone spec", func() {
+			BeforeEach(func() {
+				rs.Spec.Rclone = &scribev1alpha1.ReplicationSourceRcloneSpec{
+					ReplicationSourceVolumeOptions: scribev1alpha1.ReplicationSourceVolumeOptions{},
+					RcloneConfigSection:            &configSection,
+					RcloneDestPath:                 &destPath,
+					RcloneConfig:                   &rcloneSecret.Name,
+				}
+			})
+			When("The Job Succeeds", func() {
+				JustBeforeEach(func() {
+					Eventually(func() error {
+						return k8sClient.Get(ctx, utils.NameFor(job), job)
+					}, maxWait, interval).Should(Succeed())
+					// just so the tests will run for now
+					job.Status.Succeeded = 1
+					Expect(k8sClient.Status().Update(ctx, job)).To(Succeed())
+				})
+				When("copyMethod of None is specified for Rclone", func() {
+					BeforeEach(func() {
+						rs.Spec.Rclone.ReplicationSourceVolumeOptions.CopyMethod = scribev1alpha1.CopyMethodNone
+					})
+
+					When("No schedule is provided to ReplicationSource", func() {
+						It("NextSyncTime is never set", func() {
+							Consistently(func() bool {
+								// replication source should exist within k8s cluster
+								Expect(k8sClient.Get(ctx, utils.NameFor(rs), rs)).To(Succeed())
+								if rs.Status == nil || rs.Status.NextSyncTime.IsZero() {
+									return false
+								}
+								return true
+							}, duration, interval).Should(BeFalse())
+						})
+					})
+
+					When("Schedule is provided", func() {
+						var schedule string
+						When("Schedule is a proper cron format", func() {
+							BeforeEach(func() {
+								schedule = "1 3 3 7 *"
+								rs.Spec.Trigger = &scribev1alpha1.ReplicationSourceTriggerSpec{
+									Schedule: &schedule,
+								}
+							})
+							It("the next sync time is set in Status.NextSyncTime", func() {
+								Eventually(func() bool {
+									Expect(k8sClient.Get(ctx, utils.NameFor(rs), rs)).To(Succeed())
+									if rs.Status == nil || rs.Status.NextSyncTime.IsZero() {
+										return false
+									}
+									return true
+								}, maxWait, interval).Should(BeTrue())
+							})
+						})
+					})
+
+					It("Uses the Source PVC as the sync source", func() {
+						Eventually(func() error {
+							return k8sClient.Get(ctx, utils.NameFor(job), job)
```

Comment:
Isn't this a race w/ the controller that will delete the (completed) job?

Reply:
Not sure. It still passes when `k8sClient.Get(ctx, utils.NameFor(job), job)` is ran `Consistently` for 60 seconds.

Reply:
Looking at this a bit more...
Up at L522, you set the job to be completed. Assuming I'm following the tests correctly, I think the controller should delete the completed job. But there's no schedule defined, so I guess it's getting restarted (a 2nd sync iteration). That should be ok; it's just not what I expected.

Reply:
Moving this test block into one with a provided schedule causes the `Consistently` block to no longer find the job after a certain point, which sounds to be more in line with the behavior you expected.

Reply:
Since providing the `ReplicationSource` with a schedule causes it to be deleted, should I leave the test where it is and replace the `Eventually` block with an `Expect`?

Reply:
I think it's probably clearer to move it someplace where the job isn't marked to succeed.
Since we're just checking the job spec and don't need it to complete + get recreated, it'll be better for understanding later to have fewer moving parts in the test.